src/test/java/com/google/sps/LoginStatusTest.java

```diff
@@ -0,0 +1,156 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.common.collect.ImmutableMap;
+import com.google.gson.Gson;
+import com.google.sps.data.LoginStatus;
+import com.google.sps.servlets.LoginStatusServlet;
+import java.io.*;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import javax.servlet.http.*;
+import org.junit.Assert;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentCaptor;
+import static org.mockito.Mockito.*;
+
+@RunWith(JUnit4.class)
+public final class LoginStatusTest {
+    private static final String USER_EMAIL = "tutorguy@gmail.com";
+    private static final String USER_ID = "awesomeID";
+
+    private final LocalServiceTestHelper helper = 
+        new LocalServiceTestHelper(new LocalUserServiceTestConfig(), new LocalDatastoreServiceTestConfig())
+            .setEnvEmail(USER_EMAIL)
+            .setEnvAuthDomain("gmail.com")
+            .setEnvAttributes(
+              new HashMap(
+                  ImmutableMap.of(
+                      "com.google.appengine.api.users.UserService.user_id_key", USER_ID)));
+    
+    private HttpServletRequest request;
+    private HttpServletResponse response;
+    private LoginStatusServlet servlet;
+
+    @Before 
+    public void setUp() {
+        helper.setUp();
+
+        request = mock(HttpServletRequest.class);       
+        response = mock(HttpServletResponse.class);
+        servlet = new LoginStatusServlet();
+    }
+
+    @After
+    public void tearDown() {
+        helper.tearDown();
```

Comment:
What does this do?

Reply:
This tears down the execution environment for the local services I used. I think it's so in-memory data gets wiped out at the end of each test.

Reply:
Oh okay, that makes sense!